internal_event_handlers.go

```diff
@@ -633,23 +667,37 @@ func (weh *workflowExecutionEventHandlerImpl) handleMarkerRecorded(
 	eventID int64,
 	attributes *m.MarkerRecordedEventAttributes,
 ) error {
-	if attributes.GetMarkerName() != sideEffectMarkerName {
+	switch attributes.GetMarkerName() {
+	case sideEffectMarkerName:
+		dec := gob.NewDecoder(bytes.NewBuffer(attributes.GetDetails()))
+		var sideEffectID int32
+
+		if err := dec.Decode(&sideEffectID); err != nil {
+			return fmt.Errorf("failure decoding sideEffectID: %v", err)
+		}
+		var result []byte
+		if err := dec.Decode(&result); err != nil {
+			return fmt.Errorf("failure decoding side effect result: %v", err)
+		}
+		weh.sideEffectResult[sideEffectID] = result
+		return nil
+	case versionMarkerName:
+		dec := gob.NewDecoder(bytes.NewBuffer(attributes.GetDetails()))
+		var changeID string
+
+		if err := dec.Decode(&changeID); err != nil {
+			return fmt.Errorf("failure decoding version changeID: %v", err)
+		}
+		var version int
```

Comment:
can you use var version Version, then you don't need to cast type after extracted.